see_profile/__init__.py

```diff
@@ -0,0 +1,91 @@
+import cProfile
+import logging
+import pstats
+import sqlparse
+from io import StringIO
+
+from django.core.exceptions import MiddlewareNotUsed
+from django.db import connection
+
+from security.middleware import BaseMiddleware
+
+
+class ProfilingMiddleware(BaseMiddleware):
+    """
+    Adds the ability to profile requests via a header.
+
+    Usage:
+    Add the middleware to the MIDDLEWARE list. New boolean setting
+    "ENABLE_PROFILING" will be required to be set in the settings file. When
+    set to False, the middleware will deactivate itself. When set to True, the
+    middleware will be active.
+
+    When the middleware is active, it will log the data for any request that
+    supplies the X-Profile header in the HTTP request. This data will be logged
+    to the 'profiling' logger, so in order to see the results of this profiling
+    the Django logging will need to configure handlers for the 'profiling'
+    logger. Profiling will be configured at the DEBUG level.
+    """
+
+    REQUIRED_SETTINGS = ('ENABLE_PROFILING', 'DEBUG')
+
+    def __init__(self, get_response=None):
+        super().__init__(get_response)
+        if not self.enable_profiling:
+            raise MiddlewareNotUsed()
+        self.logger = logging.getLogger('profiling')
+
+    def load_setting(self, setting, value):
+        setattr(self, setting.lower(), value)
+
+    def format_queries(self, queries):
+        separater = f"\n{'*' * 80}\n"
+        return separater.join(
+            sqlparse.format(query['sql'], reindent=True, keyword_case='upper')
+            for query in queries
+        )
+
+    request_separator = f"\n{'=' * 80}\n"
```

Comment:
Move this up to sit alongside the other constant?

Reply:
Done.